pinhead-client/pinhead-client-spec.yaml

```diff
@@ -29,6 +29,13 @@ paths:
         schema:
           type: string
         description: "Paging offset."
+      - name: last_checkin_after
+        in: query
+        required: false
+        schema:
+          type: string
+          pattern: '^([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])T(0[0-9]|1[0-9]|2[1-4]):(0[0-9]|[1-5][0-9]):(0[0-9]|[1-5][0-9]))Z$'
```

Comment:
I am not a fan either, but the Pinhead code doesn't do ISO 8601 quite right.  For example, their code will parse `2012-12-18T23:56:23Z` but not `2012-12-18T19:56:23-04:00` which is equivalent but has an offset.  If we ever mess up and send in a time with an offset, their code just ignores the parameter: 

```
// If last_checkin_after param in the request is present but is of invalid format then the query
// for cp_consumers is not filtered by lastCheckin.
```

Ideally, they would fix their code to use a date parsing library to parse the `last_check_after` parameter instead of matching against a regex.  They could also expand the regex to allow a `Z|[+-]\d\d:\d\d` on the end, for example, but even then [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC) allows for "+5" or "+430" and while "+00:00" is allowed, "-00:00" is not.

Doing it via regex is going to leave edge cases, so my thinking was, better we catch them in our code and error out that pass on bad information and Pinhead doesn't tell us about it.

Reply:
Fair enough. :+1: